src/components/AddBookForm.js

```diff
@@ -1,11 +1,36 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
 
 function AddBookForm() {
+  const [bookData, createBook] = useState({});
+  const { titleEntry, authorEntry } = bookData;
+  const newBook = () => createBook((bookData) => {
+    const book = {
+      title: bookData.titleEntry,
+      author: bookData.authorEntry,
+      id: uuidv4(),
+    };
+    return book;
+  });
+  const dispatch = useDispatch();
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addBook(
+      newBook() || {
+        title: e.target.title.value,
+        author: e.target.author.value,
+        id: uuidv4(),
+      },
+    ));
+    e.target.reset();
+  };
   return (
-    <form>
+    <form onSubmit={handleSubmit}>
       <h2>ADD NEW BOOK</h2>
-      <input type="text" id="titleInput" placeholder="Book title" />
-      <input type="text" id="authorInput" placeholder="Author" />
+      <input type="text" id="titleInput" required name="title" placeholder="Book title" value={titleEntry} />
+      <input type="text" id="authorInput" required name="author" placeholder="Author" value={authorEntry} />
```

Comment:
- [ ] You are yet to implement the use of local states for title and author as suggested. While it appears your inputs are defined by the bookData state, you seem to dispatch the input values directly which defeats the purpose of using the local state. You are expected to dispatch the state.

- [ ] You are also expected to set the value of the inputs using onChange

  If you are struggling with the local states and dispatch you can look at these articles where the onChange method is used properly
  -  [https://www.genui.com/resources/react-hooks-simplify](https://www.genui.com/resources/react-hooks-simplify)
  - [https://dmitripavlutin.com/controlled-inputs-using-react-hooks/](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/)